app/controllers/application_controller/miq_request_methods.rb

```diff
@@ -984,69 +984,35 @@ def workflow_instance_from_vars(req)
-        @ci_node[:addClass] = "cfme-bold-node" # Show node as different
-        @ci_kids = []
-      end
-      if @curr_tag.present? && @curr_tag == t[:name]
-        t[:children].each do |c|
-          temp = {}
-          temp[:key] = c[0].to_s
-          # only add cfme_parent_key for single value tags, need to use in JS onclick handler
-          temp[:selectable] = false
-          temp[:title] = temp[:tooltip] = c[1][:description]
-          temp[:addClass] = "cfme-no-cursor-node"
-          temp[:icon] = 'fa fa-tag'
-          if edit_mode # Don't show checkboxes/radio buttons in non-edit mode
-            if vm_tags && vm_tags.include?(c[0].to_i)
-              temp[:select] = true
-              kids_checked = true
-            else
-              temp[:select] = false
-            end
-            if @edit && @edit[:current][tag_symbol_for_workflow] != @edit[:new][tag_symbol_for_workflow]
-              # checking to see if id is in current but not in new, change them to blue OR if id is in current but deleted from new
-              if (!@edit[:current][tag_symbol_for_workflow].include?(c[0].to_i) && @edit[:new][tag_symbol_for_workflow].include?(c[0].to_i)) ||
-                 (!@edit[:new][tag_symbol_for_workflow].include?(c[0].to_i) && @edit[:current][tag_symbol_for_workflow].include?(c[0].to_i))
-                temp[:addClass] = "cfme-blue-bold-node"
-              end
-            end
-            @ci_kids.push(temp) unless @ci_kids.include?(temp)
-          else
-            temp[:hideCheckbox] = true
-            @ci_kids.push(temp) unless @ci_kids.include?(temp) || !vm_tags.include?(c[0].to_i)
-          end
-        end
-      end
-      if i == tags.length - 1 # Adding last node
-        @ci_node[:expand] = true if kids_checked
-        kids_checked = false
-        @ci_node[:children] = @ci_kids if @ci_kids.present?
-        all_tags.push(@ci_node) if @ci_kids.present?
+    # for some reason @tags is set in wf, and it is changed by map bellow which causes bugs
+    wf.instance_variable_set(:@tags, nil)
+    tags = wf.allowed_tags
+    tags.each do |cat|
+      cat[:values] = cat[:children].map do |tag|
+        { :id => tag.first, :description => tag.second[:description] }
       end
+      cat[:id] = cat[:name]
+      cat[:singleValue] = cat[:single_value]
+      cat.delete(:children)
+      cat.delete(:name)
     end
-    @all_tags_tree = TreeBuilder.convert_bs_tree(all_tags).to_json # Add ci node array to root of tree
+
+    assignments = Classification.select { |tag| vm_tags.include?(tag.id) }
+    assigned_tags = assignments.map do |tag|
+      {
+        :description => tag.parent.description,
+        :id          => tag.parent.id,
+        :values      => assignments.select { |assignment| assignment.parent_id == tag.parent_id }.map do |assignment|
+          { :description => assignment.description, :id => assignment.id }
+        end
+      }
+    end.uniq
+
+    @button_urls = {
+        :save_url   => '',
```

Comment:
This looks suspicious.. you're never saving or cancelling?

(Maybe just a comment why?)

Reply:
It shouldn't be there...